compiler/rustc_codegen_ssa/src/mir/rvalue.rs

```diff
@@ -823,6 +823,12 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
 
                 OperandRef { val: OperandValue::Immediate(val), layout: box_layout }
             }
+            mir::Rvalue::WrapUnsafeBinder(ref operand, binder_ty) => {
```

Comment:
Maybe replace those new rvalues with a transmute after the analysis MIR is done?

Reply:
Hm, now that I've evaluated the change, I'm not totally certain this is worthwhile. It replaces a tiny bit of code duplication (in codegen) with a new invariant (`Rvalue::WrapUnsafeBinder` is not reachable after analysis MIR).

We'd also need to change `CastKind::Transmute` to be valid in `Analysis(PostCleanup)` rather than today when it's only allowed in `Runtime` MIR.

Reply:
ah, I assumed we had a translation pass for the transition into runtime mir anyway

Reply:
Well we do have a set of passes in `run_runtime_lowering_passes`, but there's no good pass to put it into so I'd have to add it there. I think the code deduplication ends up not really gaining us much 🤔

Reply:
Right. It's mostly a wash, but may become useful if enough optimizations will try to make use of unsafe binder projections